lib/filter/filter-expr.c

```diff
@@ -98,10 +98,10 @@ filter_expr_unref(FilterExprNode *self)
 FilterExprNode *
 filter_expr_clone(FilterExprNode *self)
 {
-  if (self->clone)
-    {
-      return self->clone(self);
-    }
+  if (!self->clone)
+    return filter_expr_ref(self);
 
-  return filter_expr_ref(self);
-}
\ No newline at end of file
+  FilterExprNode *cloned = self->clone(self);
+  cloned->comp = self->comp;
```

Comment:
Shouldn't we also copy the `modify` field?

Reply:
We should, I agree. "Modify" is currently set by the init method of FilterRE and FilterOp, so I think their "cloning" works correctly.

I'd like to do a refactor round in a separate PR, which will hopefully make filter-clone-related things a bit cleaner.
I've also found a possible cloning bug, which affects the rate-limit/throttle filter only.

Please let me know if you think the `modify` field is not working properly, in that case we should fix that too ASAP.